File: backend/src/main/java/com/slior/dto/route/UpdateRouteRequest.java
package com.slior.dto.route;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.List;

/**
 * DTO para actualizar una ruta existente.
 */
public record UpdateRouteRequest(
        @NotBlank(message = "El nombre no puede estar vacío")
        String nombre,

        @NotNull(message = "La fecha no puede estar vacía")
        LocalDate fechaPlanificada,

        String notas,

        @NotEmpty(message = "La ruta debe tener al menos una parada")
        @Valid
        List<StopRequest> paradas
) {}

File: mobile-app/app/src/main/java/com/slior/data/remote/dto/UpdateRouteRequest.kt
package com.slior.data.remote.dto

import com.slior.data.remote.dto.StopRequestDto

/**
 * DTO para actualizar una ruta existente.
 */
data class UpdateRouteRequest(
    val nombre: String,
    val fechaPlanificada: String,
    val notas: String? = null,
    val paradas: List<StopRequestDto>
)
